lib/screens/dji_shop/screens/product_screen/widgets/slider_image.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:carousel_slider/carousel_slider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_animation/infrastructure/theme/app_images.dart';
+
+class SliderImage extends StatefulWidget {
+  const SliderImage({Key? key}) : super(key: key);
+
+  @override
+  State<SliderImage> createState() => _SliderImageState();
+}
+
+class _SliderImageState extends State<SliderImage> {
+  int _current = 0;
+
+  final CarouselController _controller = CarouselController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Flexible(
+      child: Hero(
+        tag: 'productImage',
+        child: Column(
+          children: [
+            Expanded(
+              child: CarouselSlider(
+                items: imageSliders,
+                carouselController: _controller,
+                options: CarouselOptions(
+                  onPageChanged: (index, reason) {
+                    setState(() {
+                      _current = index;
+                    });
+                  },
+                ),
+              ),
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: imgList.asMap().entries.map((entry) {
+                return GestureDetector(
+                  onTap: () => _controller.animateToPage(entry.key),
+                  child: Container(
+                    width: 12.0,
+                    height: 12.0,
+                    margin: const EdgeInsets.symmetric(
+                      vertical: 8.0,
+                      horizontal: 4.0,
+                    ),
+                    decoration: BoxDecoration(
+                      shape: BoxShape.circle,
+                      color: (Theme.of(context).brightness == Brightness.dark
+                              ? Colors.white
+                              : Colors.black)
+                          .withOpacity(_current == entry.key ? 1 : 0.4),
+                    ),
+                  ),
+                );
+              }).toList(),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+final List<String> imgList = [
```

Comment:
it should be private if you create global variable which be used only inside one screen/widget